src/Common/src/Interop/Unix/libcurl/Interop.libcurl.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class libcurl
+    {
+        [DllImport(Interop.Libraries.LibCurl, SetLastError = true)]
+        public static extern SafeCurlHandle curl_easy_init();
+
+        [DllImport(Interop.Libraries.LibCurl, SetLastError = true)]
+        public static extern void curl_easy_cleanup(
+            IntPtr handle);
+
+        [DllImport(Interop.Libraries.LibCurl, CharSet = CharSet.Ansi, SetLastError = true)]
+        public static extern int curl_easy_setopt(
+            SafeCurlHandle curl,
+            int option,
+            string value);
+
+        [DllImport(Interop.Libraries.LibCurl, SetLastError = true)]
+        public static extern int curl_easy_setopt(
```

Comment:
I was not able to find a conclusive answer. I believe if internally it attempts some "system" operations like say, file write, the errno is set. If no errno is set and yet the SetLastError attribute is used, will it cause an issue?